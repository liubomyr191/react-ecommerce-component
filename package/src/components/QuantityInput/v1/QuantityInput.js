import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import ButtonBase from "@material-ui/core/ButtonBase";
import InputAdornment from "@material-ui/core/InputAdornment";
import TextField from "@material-ui/core/TextField";
import Minus from "mdi-material-ui/Minus";
import Plus from "mdi-material-ui/Plus";

const styles = () => ({
  incrementButton: {
    backgroundColor: "#fafafa",
    boxSizing: "inherit",
    color: "#666666",
    padding: 6
  },
  quantityContainer: {
    border: "1px solid #d9d9d9",
    borderRadius: "2px",
    boxSizing: "border-box",
    overflow: "hidden",
    padding: 0
  },
  quantityInput: {
    borderLeft: "1px solid #d9d9d9",
    borderRight: "1px solid #d9d9d9",
    boxSizing: "inherit",
    color: "#3c3c3c",
    fontSize: "12px",
    maxWidth: "40px",
    textAlign: "center",
    "&:focus": {
      borderColor: "#80bdff",
      boxShadow: "0 0 0 0.2rem rgba(0,123,255,.25)"
    }
  }
});

class QuantityInput extends Component {
  static propTypes = {
    /**
     * MUI theme classes
     */
    classes: PropTypes.object,
    /**
     * On change hanlder for input
     */
    onChange: PropTypes.func,
    /**
     * Prepopulate the input's value.
     */
    value: PropTypes.number
  };

  static defaultProps = {
    classes: {},
    onChange() {}
  };

  constructor(props) {
    super(props);

    const value = props.value || 0;

    this.state = {
      initialValue: props.value,
      value
    };
  }

  handleChanged(value) {
    const { onChange } = this.props;
    onChange(value);
  }

  handleQuantityInputChange = (event) => {
    const { value } = event.target;

    const numericValue = Math.floor(Number(value));

    if (Number.isNaN(numericValue)) {
      return null;
    }

    this.setState({ value: numericValue });
    this.handleChanged(numericValue);
  };

  handleIncrementButton = () => {
    const value = this.state.value + 1;

    this.setState({ value });
    this.handleChanged(value);
  };

  handleDecrementButton = () => {
    const value = this.state.value - 1;

    if (value >= 0) {
      this.setState({ value });
      this.handleChanged(value);
    }
  };

  render() {
    const { classes: { incrementButton, quantityContainer, quantityInput } } = this.props;
    const { value } = this.state;
    return (
      <TextField
        id="addToCartQuantityInput"
        value={value}
        onChange={this.handleQuantityInputChange}
        InputProps={{
          startAdornment: (
            <InputAdornment>
              <ButtonBase
                color="default"
                variant="outlined"
                onClick={this.handleDecrementButton}
                className={incrementButton}
              >
                <Minus />
              </ButtonBase>
            </InputAdornment>
          ),
          endAdornment: (
            <InputAdornment>
              <ButtonBase
                variant="outlined"
                color="default"
                onClick={this.handleIncrementButton}
                className={incrementButton}
              >
                <Plus />
              </ButtonBase>
            </InputAdornment>
          ),
          disableUnderline: true,
          classes: {
            root: quantityContainer,
            input: quantityInput
          }
        }}
      />
    );
  }
}

export default withStyles(styles)(QuantityInput);
