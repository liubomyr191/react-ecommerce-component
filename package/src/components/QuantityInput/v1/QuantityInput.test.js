import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import QuantityInput from "./QuantityInput";

test("basic snapshot", () => {
  const component = renderer.create(<QuantityInput />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
