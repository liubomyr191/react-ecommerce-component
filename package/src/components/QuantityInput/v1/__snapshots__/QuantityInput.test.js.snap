// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  aria-describedby={undefined}
  className="MuiFormControl-root-4"
>
  <div
    className="MuiInputBase-root-21 MuiInput-root-8 QuantityInput-quantityContainer-2 MuiInputBase-formControl-22 MuiInput-formControl-9 MuiInputBase-adornedStart-25 MuiInputBase-adornedEnd-26"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root-38"
    >
      <button
        className="MuiButtonBase-root-42 QuantityInput-incrementButton-1"
        color="default"
        disabled={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
        variant="outlined"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-45"
          color={undefined}
          focusable="false"
          role="presentation"
          style={
            Object {
              "fontSize": "14px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H5V11H19V13Z"
          />
        </svg>
      </button>
    </div>
    <input
      aria-invalid={false}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInputBase-input-31 MuiInput-input-16 QuantityInput-quantityInput-3 MuiInputBase-inputAdornedStart-36 MuiInputBase-inputAdornedEnd-37"
      defaultValue={undefined}
      disabled={false}
      id="addToCartQuantityInput"
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder={undefined}
      readOnly={undefined}
      required={false}
      rows={undefined}
      type="text"
      value={0}
    />
    <div
      className="MuiInputAdornment-root-38"
    >
      <button
        className="MuiButtonBase-root-42 QuantityInput-incrementButton-1"
        color="default"
        disabled={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
        variant="outlined"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-45"
          color={undefined}
          focusable="false"
          role="presentation"
          style={
            Object {
              "fontSize": "14px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
